spec/spec_helper_acceptance.rb

```diff
@@ -29,35 +29,65 @@
+          if host['platform'] =~ /^el/
+            on(host, 'yum install -y zlib-devel patch gcc-c++')
+          elsif host['platform'] =~ /^ubuntu|^debian/
+            on(host, 'apt-get install -y zlib1g-dev patch g++')
+          end
 
-      path = host.file_exist?("#{host['privatebindir']}/gem") ? host['privatebindir'] : host['puppetbindir']
-      on(host, "#{path}/gem install azure_mgmt_compute azure_mgmt_network azure_mgmt_resources azure_mgmt_storage hocon retries azure")
-    end
-  end
+          gem_command = if host['platform'] =~ /^windows/
+                          'cmd.exe /c cmd.exe /c "C:\Program Files\Puppet Labs\puppet\sys\ruby\bin\gem.bat"'
+                        else
+                          host.file_exist?("#{host['privatebindir']}/gem") ? "#{host['privatebindir']}/gem" : "#{host['puppetbindir']}/gem"
+                        end
+          if host['platform'] =~ /^windows/
+            # Unf has a separate gem for it's native extensions, unf_ext. Unf_ext has windows packages with the precompiled
+            # libraries. Because of this setup just installing the top level azure gems doesn't always seem to do the
+            # right thing on Windows
+            on(host, "#{gem_command} install unf")
+          end
+          on(host, "#{gem_command} install azure_mgmt_compute azure_mgmt_network azure_mgmt_resources azure_mgmt_storage hocon retries azure")
+        end
+      end
 
-  proj_root = File.expand_path(File.join(File.dirname(__FILE__), '..'))
-  hosts.each do |host|
-    # set :target_module_path manually to work around beaker-rspec bug that does not
-    # persist distmoduledir across runs with reused nodes
-    # TODO: ticket up this bug for beaker-rspec
-    install_dev_puppet_module_on(host, :source => proj_root, :module_name => 'azure', :target_module_path => '/etc/puppetlabs/code/modules')
-  end
+      proj_root = File.expand_path(File.join(File.dirname(__FILE__), '..'))
+      hosts.each do |host|
+        # set :target_module_path manually to work around beaker-rspec bug that does not
+        # persist distmoduledir across runs with reused nodes
+        # TODO: ticket up this bug for beaker-rspec
+        if host['platform'] =~ /^windows/
+          install_dev_puppet_module_on(host, :source => proj_root, :module_name => 'azure', :target_module_path => '/cygdrive/c/ProgramData/PuppetLabs/code/modules')
+        else
+          install_dev_puppet_module_on(host, :source => proj_root, :module_name => 'azure', :target_module_path => '/etc/puppetlabs/code/modules')
+        end
+      end
 
-  # Deploy Azure credentials to all hosts
-  if ENV['AZURE_MANAGEMENT_CERTIFICATE']
-    hosts.each do |host|
-      scp_to(host, ENV['AZURE_MANAGEMENT_CERTIFICATE'], '/tmp/azure_cert.pem')
+      # Deploy Azure credentials to all hosts
+      if ENV['AZURE_MANAGEMENT_CERTIFICATE']
+        if default['platform'] =~ /^windows/
+          scp_to(default, ENV['AZURE_MANAGEMENT_CERTIFICATE'], WINDOWS_AZURE_CERT)
```

Comment:
Same comment as above. Logically this is called `scp_to` once, but with a different argument. Something like:

`default['platform'] =~ /^windows/ ?  WINDOWS_AZURE_CERT :  WINDOWS_AZURE_CERT`